crates/daemon/src/monitor.rs

```diff
@@ -478,10 +481,14 @@ impl Actor {
 
         tracing::trace!("Updating status of {num_transactions} transactions",);
 
-        let histories = self
-            .client
-            .batch_script_get_history(self.state.monitoring_scripts())
-            .context("Failed to get script histories")?;
+        let scripts = self.state.monitoring_scripts().cloned().collect();
+
+        let mut rx_script_history = batch_script_get_history(self.client.clone(), scripts);
+
+        let mut histories = Vec::new();
```

Comment:
we can use `with_capacity(scripts.len())` here to avoid unnecessary reallocations of this vector as it grows - we know exactly the max size of the results we're gathering.

Reply:
Also done in https://github.com/itchysats/itchysats/pull/3197/commits/f8532aeff0a7a1a316a8b03110a9539d36ef387d